micronota/bfillings/_base.py

```diff
@@ -0,0 +1,83 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from abc import ABCMeta, abstractmethod
+
+from skbio import Sequence
+
+
+class IntervalMetadataPred(metaclass=ABCMeta):
```

Comment:
Because skbio.sequences contain both metadata and interval metadata.

You don't need to define two separate classes, because that is already handled internally in the skbio sequence object